### PR TITLE
finished 7 homework

### PR DESCRIPTION
**Pull request by YaroslavBelinskyi**

### CODE REVIEW

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_7/script.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+function randomEndValue() {
+    return Math.floor(Math.random() * 51 + 50);
+}
+function fibonacci() {
+    const fibonacciArray = [0, 1];
+    const endOfArray = randomEndValue();
+    for (let i = 0; ;) {
```
Comment (markoffden) [inline homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_7/script.js]: this looks odd

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_7/script.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        i = fibonacciArray[fibonacciArray.length - 1] + fibonacciArray[fibonacciArray.length - 2];
+        if (i < endOfArray) {
+            fibonacciArray.push(i);
+        } else { break; }
```
Comment (markoffden) [inline homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_7/script.js]: if you have to do thing like this a loop body, it means something may be designed better

**Comment by markoffden** [comment]
tests are cool, good job

but your function itself is very strange, I would redesign a loop

File: homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_7/script.test.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    });
+    test('Check that the array is Fibonacci sequence', () => {
+        const array = fibonacci();
+        const a = array[array.length - 1];
+        const b = array[array.length - 2];
+        const c = array[array.length - 3];
+        const cof1 = (b / a).toFixed(3);
+        const cof2 = (c / a).toFixed(3);
+        expect(cof1).toBe('0.618');
+        expect(cof2).toBe('0.382');
```
Comment (markoffden) [inline homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_7/script.test.js]: here your approach is cool, but we should test for whole array to follow Fibo secuence, that said we should test each element of array, it means you should use some kind of loop

Comment (YaroslavBelinskyi) [reply homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_7/script.test.js]: I think this is the most valid loop for this case

**Comment by markoffden** [comment]
accepted, cool approach

but I would better use loop to check each element of array to equal the sum of two previous elements

**Pull request merged**